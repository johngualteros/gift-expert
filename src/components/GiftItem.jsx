import { PropTypes } from 'prop-types'

// eslint-disable-next-line react/prop-types
function GiftItem({title, url}) {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GiftItem