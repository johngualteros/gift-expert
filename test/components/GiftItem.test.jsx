// import { checkPropTypes } from "prop-types";
import GiftItem from "../../src/components/GiftItem";
import { render, screen } from "@testing-library/react";

/* eslint-disable no-undef */
describe('Giftitem Component', () => { 
    const title = 'saitama';
    const url = 'https://test.com';
    test('should match the snapshot', () => {
        const { container } = render(<GiftItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });
    test('should render the image and alt indicated', () => {
        render(<GiftItem title={title} url={url}/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url + '/');
        expect( alt ).toBe(title);
    });
    test('should render the title indicated', () => {
        render(<GiftItem title={title} url={url}/>);
        const titleRender = screen.getByText(title);
        expect( titleRender ).toBeTruthy();
    });
 })