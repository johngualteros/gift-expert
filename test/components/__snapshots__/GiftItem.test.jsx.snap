// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Giftitem Component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://test.com"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
